src/lib.rs

```diff
@@ -1,90 +1,112 @@
 //! # Smart Leds Matrix
-//! 
+//!
 //! This is a library that adapts [smart-leds](https://crates.io/crates/smart-leds) driver implementations to the
 //! [embedded-graphics](https://docs.rs/embedded-graphics/latest/embedded_graphics/) crate by wrapping the LED
 //! driver into a `Drawable` display target.
-//! 
+//!
 
 #![no_std]
 
-use embedded_graphics_core::{Pixel, draw_target::DrawTarget, geometry::Size, geometry::{OriginDimensions}, pixelcolor::*, prelude::{Point}};
-use display_interface::DisplayError;
+use core::marker::PhantomData;
 
-use smart_leds::{SmartLedsWrite, brightness, hsv::RGB8};
+use embedded_graphics_core::{
+    draw_target::DrawTarget,
+    geometry::{OriginDimensions, Point, Size},
+    pixelcolor::{Rgb888, RgbColor},
+    Pixel,
+};
+
+use smart_leds::{brightness, hsv::RGB8, SmartLedsWrite};
 
 struct Content<const W: usize, const H: usize>(pub [[RGB8; W]; H]);
 
-impl <const W: usize, const H: usize> Content<W, H> {
+impl<const W: usize, const H: usize> Content<W, H> {
     /// Return a slice that aliases the same memory.
     pub fn as_slice(&self) -> &[RGB8] {
         // NOTE(unsafe): Creates a shared reference to the same underlying data,
         // NOTE(unsafe): which we know is tightly packed and so we can compute how many RGB8 pixel is in there.
-        unsafe { core::slice::from_raw_parts(self as *const _ as *const RGB8,
-                                             core::mem::size_of::<Self>() / core::mem::size_of::<RGB8>()) }
+        unsafe {
+            core::slice::from_raw_parts(
+                self as *const _ as *const RGB8,
+                core::mem::size_of::<Self>() / core::mem::size_of::<RGB8>(),
+            )
+        }
     }
 }
 
 /// The wrapper for the LED driver.
 ///
 /// This receives the `SmartLedsWriter` trait implementations along with a
-/// `MatrixType` that describes the size and the pixels mapping between the LED
+/// `Transformation` that describes the pixels mapping between the LED
 /// strip placement and the matrix's x y coordinates.
-pub struct SmartLedMatrix<T, M: MatrixType, const W: usize, const H: usize> {
+pub struct SmartLedMatrix<T, M: Transformation<W, H>, const W: usize, const H: usize> {
     writer: T,
     content: Content<W, H>,
-    matrix_type: M,
+    transformation: PhantomData<M>,
     brightness: u8,
 }
 
-impl<T, M: MatrixType, const W: usize, const H: usize> SmartLedMatrix<T, M, W, H> {
+impl<T, M: Transformation<W, H>, const W: usize, const H: usize> SmartLedMatrix<T, M, W, H> {
     pub fn set_brightness(&mut self, new_brightness: u8) {
         self.brightness = new_brightness;
     }
+    pub fn get_brightness(&mut self) -> u8 {
```

Comment:
The API guidelines suggest that getters should't use the `get_` prefix: https://rust-lang.github.io/api-guidelines/naming.html#getter-names-follow-rust-convention-c-getter
```suggestion
    pub fn brightness(&mut self) -> u8 {
```